httphandler/handlerequests/v1/requestshandlerutils.go

```diff
@@ -79,6 +79,17 @@ func scan(ctx context.Context, scanInfo *cautils.ScanInfo, scanID string) (*repo
 	if err := result.HandleResults(ctx); err != nil {
 		return nil, err
 	}
+	storage := storage.GetStorage()
+	if storage != nil {
+		pr := result.GetResults()
+
+		if err := storage.StorePostureReportResults(ctx, pr); err != nil {
+			return nil, err
+		}
+	} else {
+		logger.L().Debug("storage is not initialized - skipping storing results")
```

Comment:
@amirmalka  We should consider the behavior if the data is not submitted and the storage is not available, we should wait for the storage to be available before scanning

Reply:
@dwertent let's discuss this, As I understand from Vlad it involves implementing a retry mechanism (which is something we didn't yet implemented in the other services)